src/main/java/io/quarkus/optimus/CustomRepositoryConnectorFactory.java

```diff
@@ -123,8 +123,17 @@ public void get(Collection<? extends ArtifactDownload> artifactDownloads, Collec
                             modifyPomFile(toTransform, entry.getValue());
                             Matcher m = Pattern.compile("<version>(.*?)</version>").matcher(pomFile);
                             StringBuffer sb = new StringBuffer();
+                            StringBuffer lineBuf = new StringBuffer();
+                            int indx = 0;
                             while (m.find()) {
-                                m.appendReplacement(sb, "<version>$1" + "-\\$\\$jakarta9\\$\\$</version>");
+                                m.appendReplacement(lineBuf, "<version>$1" + "-\\$\\$jakarta9\\$\\$</version>");
+                                if ((indx = lineBuf.toString().lastIndexOf("<version>${project.version}-$$jakarta9$$</version>")) > 0) {
+                                    sb.append(lineBuf.toString().substring(0, indx-1));
+                                    sb.append("<version>${project.version}</version>");
```

Comment:
I don't really understand what this is doing? Is just not doing the replacement if the match is ${project.version}? 

If so why not just use `if(m.group(1).equals("${project.version}")))` sb.append(m.group(0)); else m.appendReplacement(sb, "<version>$1" + "-\\$\\$jakarta9\\$\\$</version>");`

Reply:
> Is just not doing the replacement if the match is ${project.version}?
Yes.

I am not that familiar with how groups work, so if your way works better I say go for it.